### ISSUE TITLE
Can't add to empty dice-box

### ISSUE THREAD

**Issue by CapsE** [3d-dice/dice-box-threejs#4]
I know it's not really a bug but it would be nice to be able to simply always call "add" instead of having to check if a roll was already made. I fixed this by adding
`if (!prevNotation) return newNotation;` to DiceNotation.js:202 This seems to solve the issue.

Currently all dice get removed when calling "add" while a roll is still going on. I'm guessing that is needed for the deterministic rolling? It would be nice to be able to just keep adding dice. Maybe it could work when you disable collisions between dice that weren't added in the same roll? That would keep each roll deterministic while preventing dice from being deleted.

**Comment by frankieali** [comment]
It is possible to add more dice to a roll, but you are correct that dice can not be added while a previous roll is still rolling due to determinism.

I use this code in my test app.
```javascript
  Box.roll("1d10+1d100+1d2+1d6").then(result=>{
    console.log('result :>> ', result);
    Box.reroll([0,1]).then(result=>{
      console.log('result :>> ', result);
      Box.add("3d6@2,2,2").then(result=>{
        console.log('result :>> ', result);
        Box.remove([1,5]).then(result => {
          console.log('result :>> ', result);
          console.log('Box.getDiceResults() :>> ', Box.getDiceResults());
        })
      })
    })
  })
  ```

**Comment by CapsE** [comment]
I know this is kind of two things in one now. Have you considered my fix for the initial problem? I feel like calling add should not run in an unhandled error if it's done before the first roll. The suggested changes makes it so it just works.

Regarding adding before a roll is finished: I'm building a network app so multiple users could trigger an "add" or "roll" at any point. I'd like to use add so no rolls get lost. I guess I could write a queue that collects rolls to add and runs them one after the other. Would that be something that should be done by the add method automatically? I get why it has to clear the dice but it still is unexpected when calling the method especially because it's also clearing dice that are no longer rolling.

**Comment by frankieali** [comment]
So typically a roll is started with the `.roll()` method. The `.add()` method is used to add additional dice to the roll. The dice rolled previously become static rigid bodies so their outcome will not change due to collisions. I think building a queue is probably the only way to get that working. 
The way the determinism works is the simulation is run very rapidly in code to determine the outcome of the dice. Then the simulation is given all the same initial starting values with the exception that now the dice face values have been swapped to the desired result. The simulation runs a second time and is now synced with the rendering engine. Adding dice cannot be done in mid-simulation.

**Comment by frankieali** [comment]
the `.add()` method has been updated to call `.roll()` if there are no dice in the box. This is available in version `0.0.12`

**Comment by CapsE** [comment]
Thanks, it works! It's a bit weirder now when adding dice while a roll is still going on but I'll just add a queue to avoid that.
